### PR TITLE
Version 1.0 with processors

### PR DESCRIPTION
**Pull request by s12chung**
see: https://github.com/FinalCAD/simple_controller/tree/extension#post-process

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,9 +70,66 @@ Router.call("threes/multiply", number: 6) # calls ThreesController.call(:multipl
...
+  def readme
+    write_new_file_and_return_path
+  end
+end
+```
+
+And the [Router](#router) is set up, `FoldersController#two_readmes` generates a directory of `FilesController#readme`s. Processors add the ability to do these calls:
+
+```ruby
+Router.call('files/readme.s3_key')
```
Comment (joel) [inline README.md:110]: `s3` is a named service of Amazon but the concept is more general of that, maybe we can use only `key` or `remote_key` to stay abstract?

File: spec/simple_controller/base_spec.rb [hunk 2]
```diff
@@ -29,12 +40,67 @@
...
+          end
+        end
+      end
+
+      describe "#params" do
+        let(:params) { { number: 3, controller: "api/numbers" } }
+        subject { instance.params }
+
+        it "works" do
+          expect(subject).to eql ActiveSupport::HashWithIndifferentAccess.new(params)
```
Comment (joel) [inline spec/simple_controller/base_spec.rb:93]: `expect(subject).to eql(params.with_indifferent_access)` Should be the same

Comment (s12chung) [inline spec/simple_controller/base_spec.rb:93]: forgot :p, you're right, it's better

File: README.md (hunk 1 above)
Comment (s12chung) [inline README.md:110]: well... this is just example code. the `post_process` method defines the extensions.

Comment (s12chung) [inline README.md:110]: i should probably explicitly state that they're defined below.

File: spec/simple_controller/base_spec.rb (hunk 2 above)
Comment (s12chung) [inline spec/simple_controller/base_spec.rb:93]: didn't work, probably need to include stuff. rather not.

**Pull request merged**